templates/JSResolverOCHTTPS.js

```diff
@@ -64,11 +64,17 @@ export function resolveGraphDBQueryFromEvent(event) {
         const value = event.arguments[inputDef.name.value];
 
         if (value) {
-            const inputType = typeFromAST(schema, inputDef.type);
+            let inputType = typeFromAST(schema, inputDef.type);
+            if (inputType instanceof GraphQLNonNull && inputType.ofType) {
+                // if non-null type, unwrap to the underlying type
+                inputType = inputType.ofType;
+            }
             const astValue = astFromValue(value, inputType);
             if (inputType instanceof GraphQLInputObjectType) {
                 // retrieve an ID field which may not necessarily be named 'id'
-                const idField = Object.values(inputType.getFields()).find(field => field.type.name === GraphQLID.name);
+                const idField = Object.values(inputType.getFields()).find(
+                    // if field is of non-null type, check the underlying type as well
+                    field => field?.type?.name === GraphQLID.name || field?.type?.ofType?.name === GraphQLID.name);
```

Comment:
```suggestion
                    field => field.type?.name === GraphQLID.name || field.type?.ofType?.name === GraphQLID.name);
```

I might be wrong, but I don't think `field` can ever be undefined or null here.